NETCORE/src/Shared/Internals/ApplicationNameProvider.cs

```diff
@@ -25,7 +25,17 @@ private static string GetApplicationName()
         {
             try
             {
-                return Assembly.GetEntryAssembly().GetName().Name;
+                var assembly = Assembly.GetEntryAssembly();
```

Comment:
Is this related to heartbeat?

Reply:
no, not related. This throws NullRefExceptions and was making it difficult to debug.